pkg/cgroups/cgroups.go

```diff
@@ -0,0 +1,155 @@
+package cgroups
+
+import (
+	"bufio"
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+// map of cgroup subsystems to their respective paths
+type Cgroups map[string]string
+
+// Find retrieves the mount points for all cgroup subsystems on the host. The
+// result of this operation should be cached if possible.
+func Find() (Cgroups, error) {
+	// refer to `man proc` or
+	// https://www.kernel.org/doc/Documentation/filesystems/proc.txt section 3.5
+	mountInfo, err := os.Open("/proc/self/mountinfo")
+	if err != nil {
+		return nil, err
+	}
+	defer mountInfo.Close()
+
+	cgroupSystems, err := os.Open("/proc/cgroups")
+	if err != nil {
+		return nil, err
+	}
+	defer cgroupSystems.Close()
+
+	var subsystems []string
+	subsysScanner := bufio.NewScanner(cgroupSystems)
+	for subsysScanner.Scan() {
+		lineSegs := strings.Fields(subsysScanner.Text())
+		subsystems = append(subsystems, lineSegs[0])
+	}
+
+	ret := make(map[string]string)
+
+	scanner := bufio.NewScanner(mountInfo)
+	for scanner.Scan() {
+		lineSegs := strings.Fields(scanner.Text())
+		if lineSegs[len(lineSegs)-3] != "cgroup" {
+			// filesystem type is not "cgroup", skip
+			continue
+		}
+		mountPoint := lineSegs[4]
+
+		cgroupSuperOpts := strings.Split(lineSegs[len(lineSegs)-1], ",")
+		for _, opt := range cgroupSuperOpts {
+			for _, subsystem := range subsystems {
+				if opt == subsystem {
+					ret[subsystem] = mountPoint
+					break
+				}
+			}
+		}
+	}
+
+	return ret, nil
+}
+
+var Default Cgroups = map[string]string{
+	"cpu":    "/cgroup/cpu",
```

Comment:
There are a lot of subsystems, just we happen to only use these two... I was considering a refactor to struct though and it's probably the right call. Will change